module/zfs/dbuf.c

```diff
@@ -975,10 +975,11 @@ dbuf_verify(dmu_buf_impl_t *db)
 		ASSERT3U(db->db.db_offset, ==, db->db_blkid * db->db.db_size);
 	}
 
-	for (dr = db->db_data_pending; dr != NULL; dr = dr->dr_next)
+	for (dr = db->db_data_pending; dr != NULL;
+	    dr = list_next(&db->db_dirty_records, dr))
 		ASSERT(dr->dr_dbuf == db);
-
-	for (dr = db->db_last_dirty; dr != NULL; dr = dr->dr_next)
+	for (dr = list_head(&db->db_dirty_records);
+	    dr != NULL; dr = list_next(&db->db_dirty_records, dr))
```

Comment:
Doesn't think end up looping over a portion of the list, then looping over the entire list?

Am I misunderstanding what this is doing?

Reply:
do we know that they're both the same list?

Reply:
we are calling list_next() on the same list in both cases, that is the part I wasn't sure was correct.

Reply:
list_next is just an accessor to calculate the offset of the list entry for traversal